onnxscript/rewriter/llama_rule_sets.py

```diff
@@ -32,26 +32,23 @@ def check(self, context, x) -> orp.MatchResult:
         return check_result
 
 
-class CastIdentity(orp.RewriteRuleAsClass):
+class CastIdentity(orp.RewriteRuleClassBase):
     """Replaces ``Cast(., to=to)`` by ``Identity`` if possible."""
 
-    @classmethod
-    def pattern(cls, op, x, to):
+    def pattern(self, op, x, to):
         return op.Cast(x, to=to)
 
-    @classmethod
-    def rewrite(cls, op, x: ir.Value, to: ir.Attr):
+    def rewrite(self, op, x: ir.Value, to: ir.Attr):
         return op.Identity(x)
 
-    @classmethod
-    def check(cls, context, x, to) -> orp.MatchResult:
+    def check(self, context, x, to) -> orp.MatchResult:
         check_result = orp.MatchResult()
-        if x.dtype != to.value:
+        if x.dtype != to.as_int():
```

Comment:
Trying to understand this: does `as_int` throw an exception if it is not an int attribute? If so, the existing version seems preferable?

One design question worth thinking about is: what the optimization rules should do when dealing with invalid nodes: eg., if we have a model with a Cast node where the "to" attribute is a string or float. The comparison with `to.value` will choose to not optimize the node, but not fail, while the changed version might fail.

Reply:
>  does as_int throw an exception if it is not an int attribute
I think so.

Good point thanks. I was assuming the graph to be valid (can pass checker). I suppose we can assume graphs to be always valid in rewrites, other wise moving responsibilities of ensuring graph validity to the rules may be burdensome?

Reply:
Will add the explicit check to as_int() as discussed